### PR TITLE
New fetch issue fix without any reformatting

### PR DESCRIPTION
**Pull request by ThomasJejkal**
Applied logical changes from former fetch_fix branch again without changing any formatting

### CODE REVIEW

**Comment by ThomasJejkal** [comment]
I've no idea why the build fails. On my local machines, everything runs fine and the code changes are the same as before, where the build succeeded. Unfortunately, there is no way for me to check what exactly fails during the remote build.

File: src/main/java/gov/loc/repository/bagit/writer/PayloadWriter.java [hunk 1]
```diff
@@ -83,11 +83,14 @@ public static void writePayloadFiles(final Set<Manifest> payloadManifests, final
...
- private static Set<Path> getFetchPaths(final List<FetchItem> fetchItems){
-   final Set<Path> fetchPaths = new HashSet<>();
-   for(final FetchItem fetchItem : fetchItems) {
-     fetchPaths.add(fetchItem.getPath());
-   }
-   return fetchPaths;
+  private static Set<Path> getFetchPaths(final List<FetchItem> fetchItems, final Path bagDataDir) {
+    final Set<Path> fetchPaths = new HashSet<>();
+    for (final FetchItem fetchItem : fetchItems) {
+      Path parent = bagDataDir.getParent();
```
Comment (johnscancella) [inline src/main/java/gov/loc/repository/bagit/writer/PayloadWriter.java]: PMD is complaining that `parent` could be declared final.

File: src/main/java/gov/loc/repository/bagit/writer/PayloadWriter.java [hunk 2]
```diff
@@ -61,7 +61,7 @@ static Path writeVersionDependentPayloadFiles(final Bag bag, final Path outputDi
   */
  public static void writePayloadFiles(final Set<Manifest> payloadManifests, final List<FetchItem> fetchItems, final Path outputDir, final Path bagDataDir) throws IOException{
    logger.info(messages.getString("writing_payload_files"));
-   final Set<Path> fetchPaths = getFetchPaths(fetchItems);
+    final Set<Path> fetchPaths = getFetchPaths(fetchItems, bagDataDir);
```
Comment (johnscancella) [inline src/main/java/gov/loc/repository/bagit/writer/PayloadWriter.java]: please remove extra space here

**Comment by johnscancella** [comment]
I manually merged this into the master branch. Thanks for all the hard work!

**Pull request closed without merging**